src/sentry/lang/javascript/processor.py

```diff
@@ -358,56 +359,80 @@ def fetch_file(url, project=None, release=None, allow_scraping=True):
-                    url,
-                    allow_redirects=True,
-                    verify=False,
-                    headers=headers,
-                    timeout=settings.SENTRY_SOURCE_FETCH_TIMEOUT,
-                )
-            except Exception as exc:
-                logger.debug('Unable to fetch %r', url, exc_info=True)
-                if isinstance(exc, RestrictedIPAddress):
-                    error = {
-                        'type': EventError.RESTRICTED_IP,
-                        'url': expose_url(url),
-                    }
-                elif isinstance(exc, SuspiciousOperation):
-                    error = {
-                        'type': EventError.SECURITY_VIOLATION,
-                        'url': expose_url(url),
-                    }
-                elif isinstance(exc, (RequestException, ZeroReturnError)):
-                    error = {
-                        'type': EventError.JS_GENERIC_FETCH_ERROR,
-                        'value': six.text_type(type(exc)),
-                        'url': expose_url(url),
-                    }
-                else:
-                    logger.exception(six.text_type(exc))
-                    error = {
-                        'type': EventError.UNKNOWN_ERROR,
-                        'url': expose_url(url),
-                    }
-
-                # TODO(dcramer): we want to be less aggressive on disabling domains
-                cache.set(domain_key, error or '', 300)
-                logger.warning('Disabling sources to %s for %ss', domain, 300,
-                               exc_info=True)
-                raise CannotFetchSource(error)
-
-            # requests' attempts to use chardet internally when no encoding is found
-            # and we want to avoid that slow behavior
-            if not response.encoding:
-                response.encoding = 'utf-8'
-
-            body = response.text
-            z_body = zlib.compress(force_bytes(body))
-            headers = {k.lower(): v for k, v in response.headers.items()}
-
-            cache.set(cache_key, (headers, z_body, response.status_code), 60)
-            result = (headers, body, response.status_code)
+                try:
+                    now = time.time()
+                    response = http_session.get(
+                        url,
+                        allow_redirects=True,
+                        verify=False,
+                        headers=headers,
+                        timeout=settings.SENTRY_SOURCE_FETCH_SOCKET_TIMEOUT,
+                        stream=True,
+                    )
+
+                    try:
+                        cl = int(response.headers['content-length'])
+                    except (LookupError, ValueError):
+                        cl = 0
+                    if cl > settings.SENTRY_SOURCE_FETCH_MAX_SIZE:
+                        raise OverflowError()
+                    contents = []
+                    for chunk in response.iter_content():
```

Comment:
This will be really bad. Default `chunk_size` value is `1`.

See:

https://github.com/kennethreitz/requests/blob/master/requests/models.py#L658
https://github.com/kennethreitz/requests/blob/master/requests/models.py#L757
https://github.com/kennethreitz/requests/blob/master/requests/models.py#L54